moonraker/plugins/octoprint_compat.py

```diff
@@ -0,0 +1,280 @@
+            protocol=['http_octo'])
+        self.server.register_endpoint(
+            '/api/settings', ['GET'], self._get_settings,
+            protocol=['http_octo'])
+
+        # File operations
+        # Note that file upload is handled in file_manager.py
+        # TODO: List/info/select/delete files
+
+        # Job operations
+        self.server.register_endpoint(
+            '/api/job', ['GET'], self._get_job, protocol=['http_octo'])
+        # TODO: start/cancel/restart/pause jobs
+
+        # Printer operations
+        self.server.register_endpoint(
+            '/api/printer', ['GET'], self._get_printer, protocol=['http_octo'])
+        self.server.register_endpoint(
+            '/api/printer/command', ['POST'], self._post_command,
+            protocol=['http_octo'])
+        # TODO: head/tool/bed/chamber specific read/issue
+
+        # Printer profiles
+        self.server.register_endpoint(
+            '/api/printerprofiles', ['GET'], self._get_printerprofiles,
+            protocol=['http_octo'])
+
+        # System
+        # TODO: shutdown/reboot/restart operations
+
+    async def _init(self):
+        self.klippy_apis = self.server.lookup_plugin('klippy_apis')
+        # Fetch heaters
+        try:
+            result = await self.klippy_apis.query_objects({'heaters': None})
+        except self.server.error as e:
+            logging.info(f'Error Configuring heaters: {e}')
+            return
+        self.heaters = result.get('heaters', {}).get('available_sensors', [])
+
+    async def printer_state(self):
+        if not self.klippy_apis:
+            return 'Offline'
+        if self.server.klippy_state != 'ready':
+            return 'Error'
+        result = await self.klippy_apis.query_objects({'print_stats': None})
+        return {
+            'standby': 'Operational',
+            'printing': 'Printing',
+            'paused': 'Paused',
+            'complete': 'Operational'
+        }.get(result.get('state', 'standby'), 'Error')
+
+    async def printer_temps(self):
+        temps = {}
+        if not self.klippy_apis:
+            return temps
+        if self.heaters:
+            result = await self.klippy_apis.query_objects(
+                {heater: None for heater in self.heaters})
+            for heater in self.heaters:
+                data = result[heater]
+                name = 'bed'
+                if heater.startswith('extruder'):
+                    try:
+                        tool_no = int(heater[8:])
+                    except ValueError:
+                        tool_no = 0
+                    name = f'tool{tool_no}'
+
+                temps[name] = {
+                    'actual': int(data['temperature'] * 100.0) / 100.0,
+                    'offset': 0,
+                    'target': data['target'],
+                }
+        return temps
+
+    async def _get_version(self, args):
+        """
+        Version information
+        """
+        ver = utils.get_software_version()
```

Comment:
Rather than importing utils to fetch the software version, you can get it from the config object in `__init__()`, for example:
```python
self.software_version = config['system_args'].get("software_version")
```